### PR TITLE
Fix ink! message result return reverts

### PR DESCRIPTION
**Pull request by Robbepop**
Fixes https://github.com/paritytech/ink/issues/985.

@xgreenx Please check if this fixes the problems for you.

Not yet optimized. Will do once we know for sure that this fix actually works.

### CODE REVIEW

File: crates/lang/codegen/src/generator/dispatch.rs [hunk 1]
```diff
@@ -672,18 +672,25 @@ impl Dispatch<'_> {
...
+                        mutates: #mutates_storage,
+                        dynamic_storage_alloc: #is_dynamic_storage_allocation_enabled,
+                    };
+                    let mut contract: ::core::mem::ManuallyDrop<#storage_ident> =
+                        ::core::mem::ManuallyDrop::new(
+                            ::ink_lang::codegen::initiate_message::<#storage_ident>(config)?
+                        );
+                    let result: #message_output = #message_callable(&mut contract, input);
+                    let failure = ::ink_lang::is_result_type!(#message_output)
+                        && ::ink_lang::is_result_err!(&result);
```
Comment (xgreenx) [inline crates/lang/codegen/src/generator/dispatch.rs:686]:
```suggestion
                        && ::ink_lang::is_result_err!(result);
```

**Review by xgreenx — COMMENTED** [review]
It works=)

**Comment by xgreenx** [comment]
I tested with Europa and you can see that `flags: 1`. So it is reverted
image

File: crates/lang/codegen/src/generator/dispatch.rs (hunk 1 above)
Comment (Robbepop) [reply crates/lang/codegen/src/generator/dispatch.rs]: Thank you for the debugging help!

**Comment by Robbepop** [comment]
On the downside of this PR is that it currently slightly increases the Wasm file sizes of some ink! smart contracts. E.g. ERC-20 raised from 28.9kB to 31.4kB in size. We might be able to find some optimizations to reduce this additional bloat.

**Review by cmichi — APPROVED** [review]

**Pull request merged**